core/api.go

```diff
@@ -316,6 +317,22 @@ func blockchainIDHandler(handler http.Handler, blockchainID string) http.Handler
 	})
 }
 
+// loggingHandler pulls out request data and adds it to the request's
+// logging context.
+func loggingHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		ctx = log.AddPrefixkv(ctx, "path", req.URL.Path)
+		if userAgent := req.UserAgent(); userAgent != "" {
+			ctx = log.AddPrefixkv(ctx, "useragent", userAgent)
```

Comment:
Okay yeah this seems good for now. At least if we want to change our output style, we can think about that separately.